### PR TITLE
add missing resource section for houston db migration job spec

### PR DESCRIPTION
**Pull request by Shubham-astro**
## Description

This PR introduces resource section which were missing in db migration job causing OOM errors.

## Related Issues

astronomer/issues#6598

## Testing

Done Locally

Screenshot 2024-09-03 at 12 51 42 PM

## Merging

NA

### CODE REVIEW

File: tests/chart_tests/test_astronomer_houston_hook_job.py [hunk 1]
```diff
@@ -102,3 +102,6 @@ def test_db_migration_job_defaults(self, kube_version):
         assert c_by_name["houston-db-migrations-job"]["securityContext"] == {
             "runAsNonRoot": True
         }
+	assert "resources" in c_by_name["wait-for-db"]
+    	assert "resources" in c_by_name["houston-bootstrapper"]
+    	assert "resources" in c_by_name["houston-db-migrations-job"]
```
Comment (pgvishnuram) [inline tests/chart_tests/test_astronomer_houston_hook_job.py:107]: can you also add test cases to validate resource overrides as well

File: tests/chart_tests/test_astronomer_houston_hook_job.py [hunk 2]
```diff
@@ -3,102 +3,89 @@
 from tests.chart_tests.helm_template_generator import render_chart
 
 
-@pytest.mark.parametrize(
-    "kube_version",
-    supported_k8s_versions,
-)
```
Comment (pgvishnuram) [inline tests/chart_tests/test_astronomer_houston_hook_job.py:9]: this is used to paramaterize multiple k8s version this shouldn't be removed

File: charts/astronomer/templates/houston/helm-hooks/houston-au-strategy-job.yaml [hunk 3]
```diff
@@ -52,6 +52,8 @@ spec:
           # If you supply only args for a Container, the default Entrypoint defined in the Docker image is run with the args that you supplied.
           args: ["update-deployments-resource-mode"]
           securityContext: {{ toYaml .Values.securityContext | nindent 12 }}
+          resources:
+{{ toYaml .Values.houston.resources | indent 12 }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/houston/helm-hooks/houston-au-strategy-job.yaml:56]: `nindent` adds `\n` at the beginning, and using it makes the template a bit cleaner.
```suggestion
          resources: {{ toYaml .Values.houston.resources | nindent 12 }}
```

File: charts/astronomer/templates/houston/helm-hooks/houston-db-migration-job.yaml [hunk 4]
```diff
@@ -62,6 +62,8 @@ spec:
           image: {{ template "dbBootstrapper.image" . }}
           imagePullPolicy: {{ .Values.images.dbBootstrapper.pullPolicy }}
           securityContext: {{ toYaml .Values.securityContext | nindent 12 }}
+          resources:
+{{ toYaml .Values.houston.resources | indent 12 }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/houston/helm-hooks/houston-db-migration-job.yaml:66]:
```suggestion
          resources: {{ toYaml .Values.houston.resources | nindent 12 }}
```

File: charts/astronomer/templates/houston/helm-hooks/houston-db-migration-job.yaml [hunk 5]
```diff
@@ -94,9 +96,10 @@ spec:
         - name: houston-db-migrations-job
           image: {{ template "houston.image" . }}
           imagePullPolicy: {{ .Values.images.houston.pullPolicy }}
-          # If you supply only args for a Container, the default Entrypoint defined in the Docker image is run with the args that you supplied.
           args: ["yarn", "migrate"]
           securityContext: {{ toYaml .Values.securityContext | nindent 12 }}
+          resources:
+{{ toYaml .Values.houston.resources | indent 12 }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/houston/helm-hooks/houston-db-migration-job.yaml:102]:
```suggestion
          resources: {{ toYaml .Values.houston.resources | nindent 12 }}
```

File: tests/chart_tests/test_astronomer_houston_hook_job.py [hunk 6]
```diff
@@ -94,11 +105,30 @@ def test_db_migration_job_defaults(self, kube_version):
...
+        docs_overridden = render_chart(kube_version=kube_version, values=value, 
+                                       show_only=['charts/astronomer/templates/houston/helm-hooks/houston-db-migration-job.yaml'])
+        assert len(docs_overridden) == 1
+        c_by_name_overridden = get_containers_by_name(docs_overridden[0], include_init_containers=True)
+
+        
+
+        assert c_by_name_overridden['wait-for-db']['resources'] == overridden_resources
+        assert c_by_name_overridden['houston-bootstrapper']['resources'] == overridden_resources
+        assert c_by_name_overridden['houston-db-migrations-job']['resources'] == overridden_resources
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_hook_job.py:133]: This should not be in `test_db_migration_job_defaults()`, but should be a different test function, like `test_db_migration_job_custom_resources()`

File: tests/chart_tests/test_astronomer_houston_hook_job.py [hunk 7]
```diff
@@ -94,11 +105,45 @@ def test_db_migration_job_defaults(self, kube_version):
...
+
+        assert c_by_name_overridden["wait-for-db"]["resources"] == overridden_resources
+        assert (
+            c_by_name_overridden["houston-bootstrapper"]["resources"]
+            == overridden_resources
+        )
+        assert (
+            c_by_name_overridden["houston-db-migrations-job"]["resources"]
+            == overridden_resources
+        )
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_hook_job.py:147]: Nit: if we change `c_by_name_overridden` to `c_by_name` and `overridden_resources` to `overrides` these checks will each fit on one line, and so will lines 135-137, which makes this code a lot easier to grok at a glance.

File: tests/chart_tests/test_astronomer_houston_hook_job.py [hunk 8]
```diff
@@ -94,11 +105,44 @@ def test_db_migration_job_defaults(self, kube_version):
...
+#EOF
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_hook_job.py:148]: What is the purpose of this?

File: tests/chart_tests/test_astronomer_houston_hook_job.py (hunk 6 above)
Comment (Shubham-astro) [reply tests/chart_tests/test_astronomer_houston_hook_job.py]: this is done

File: tests/chart_tests/test_astronomer_houston_hook_job.py (hunk 8 above)
Comment (Shubham-astro) [reply tests/chart_tests/test_astronomer_houston_hook_job.py]: EOF is added due to one of the checks that is failing:  fix end of files
So i was trying this out with a comment and a newline.

File: tests/chart_tests/test_astronomer_houston_hook_job.py (hunk 7 above)
Comment (Shubham-astro) [reply tests/chart_tests/test_astronomer_houston_hook_job.py]: done

File: tests/chart_tests/test_astronomer_houston_hook_job.py (hunk 8 above)
Comment (danielhoherd) [reply tests/chart_tests/test_astronomer_houston_hook_job.py]: fix-end-of-files makes sure that all whitespace is trimmed off the end of the file, with only one trailing newline. If you `pre-commit install` and then `pre-commit run --all-files` in your local checkout then it'll fix this all locally for you. The idea here is that these changes will be made when you `git commit` if anything is out of line, you can review the updates, `git add --update` and then `git commit` again.

**Review by pgvishnuram — APPROVED** [review]

**Pull request merged**